cumulus_library/studies/core/core_templates/documentreference.sql.jinja

```diff
@@ -67,8 +66,14 @@ SELECT DISTINCT
     tdr.author_year,
     tdr.doc_format_code,
     tdr.subject_ref,
-    coalesce(tdr.encounter_ref, context_encounter.encounter.reference) as encounter_ref,
+    {% if schema["documentreference"]["context"]["encounter"] %}
+    context_encounter.encounter.reference AS encounter_ref,
+    {% else %}
+    CAST(NULL AS VARCHAR) AS encounter_ref,
+    {% endif %}
```

Comment:
This approach has the benefit of not blowing up for a database without context.encounter (thus, is helpful for light-weight unit tests).

But... if the field exists in the schema, the unnest means that we don't include any DocRefs without linked encounters at all (which might be a reasonable position). But if it's not in the schema, you _do_ get NULL encounter_refs.

So this fix is a bit inconsistent. And we should either unnest as a temp table then `LEFT JOIN` in order to allow loose DocRefs or change the no-schema approach to drop unlinked rows instead of using NULL.

Reply:
I think i'm in favor of the first one.